src/packages/@ncigdc/containers/explore/ClinicalAggregations.tsx

```diff
@@ -263,41 +234,31 @@ const enhance = compose(
             aggregation.stats && aggregation.stats.count === 0,
           ])
       );
-
-      const filteredFacets = _.filter(_.values(facetMapping), facet => {
-        return _.every([
-          facetMatchesQuery(
-            facet,
-            _.get(parsedFacets[facet.field], 'buckets', undefined),
-            searchValue
-          ),
-          !facetExclusionTest(facet),
-          !shouldHideUselessFacets ||
-            Object.keys(usefulFacets).includes(facet.field),
-          !facet.field.endsWith('id'),
-          !facet.field.includes('updated_datetime'),
-          !facet.field.includes('created_datetime'),
-        ]);
-      });
-      const fieldHash = {};
-      let key = '';
-      for (const str of filteredFacets.map((f: any) => f.field)) {
-        const el = str.split('.');
-        let subFieldHash = fieldHash;
-        while (el.length >= 1) {
-          key = el.shift() || '';
-          if (el.length === 0) {
-            subFieldHash[key] = facetMapping['cases.' + str];
-          } else {
-            subFieldHash[key] = subFieldHash[key] || {};
-            subFieldHash = subFieldHash[key];
-          }
-        }
-      }
+      const filteredFacets = clinicalFacets.reduce((acc, header) => {
+        return {
+          ...acc,
+          [header.field]: _.filter(facetMapping, facet => {
+            return _.every([
+              facetMatchesQuery(
+                facet,
+                _.get(parsedFacets[facet.field], 'buckets', undefined),
+                searchValue
+              ),
+              !facetExclusionTest(facet),
+              !shouldHideUselessFacets || facet.field in usefulFacets,
+              facet.full.startsWith(header.full),
+              !header.excluded || !facet.full.startsWith(header.excluded),
+              !facet.field.endsWith('id'),
```

Comment:
> I'm thinking we should use a explicit blacklist from constants, because I think we'll be _exactly_ what we're blacklisting. Like:
> BLACKLISTED_FIELDS = ['updated_datetime', 'created_datetime', 'state', ...etc]
> What do you think?

Yes. for sure. The only problem is from the id. I can add the blacklist in the parsedFacets file as a extra column. But to for all the id(XXX_id ), I need to filter it in js like startsWith('id'). Let me think about this.

Reply:
I think whatever we do we should try to use a consistent method. I used a regex with the explicit value so I could tell the exact fields i was blacklisting, which I like, and I think there's less danger of excluding more than we want to.